driver-core/src/main/com/mongodb/operation/AsyncQueryBatchCursor.java

```diff
@@ -274,16 +299,19 @@ private BsonDocument asKillCursorsCommandDocument(final ServerCursor localCursor
                 .append("cursors", new BsonArray(singletonList(new BsonInt64(localCursor.getId()))));
     }
 
+    private void endOperationInProgress() {
+        boolean pendingClose = false;
+        synchronized (this) {
+            noOperationInProgress.set(true);
+            pendingClose = closePending.get();
+        }
+        if (pendingClose) {
+            close();
+        }
+    }
 
     private void handleGetMoreQueryResult(final AsyncConnection connection, final SingleResultCallback<List<T>> callback,
                                           final QueryResult<T> result, final boolean tryNext) {
-        if (isClosed()) {
-            connection.release();
-            callback.onResult(null, new MongoException(format("The cursor was closed before %s completed.",
-                    tryNext ? "tryNext()" : "next()")));
-            return;
-        }
-
         cursor.getAndSet(result.getCursor());
```

Comment:
Nit: this can just be `cursor.set(result.getCursor())` since we're not using the return value.

Reply:
Done